go/internal/handshake/handshake_test.go

```diff
@@ -25,7 +24,7 @@ func TestValidHandshake(t *testing.T) {
 		stream p2pnetwork.Stream,
 		mh *mockedHandshake,
 	) {
-		defer p2phelpers.FullClose(stream)
+		defer stream.Close()
```

Comment:
```suggestion
		defer stream.Reset()
```
Closes are async, it's more a free the buffers and stop saving what the other node is sending, I don't care it anyway.

To replace a `p2phelpers.FullClose` a `stream.Reset` is more suited.

Reply:
from what I understood, that's what changed in https://github.com/libp2p/go-libp2p-core/pull/166 and around

from what I understand `Close` is now the safe sync way to close a stream

can you double-check? I will adapt accordingly :+1:, thx

Reply:
PS: Now that I think about that I have never checked the behaviour of `Reset`, I guess I should this later.

Reply:
> 
> 
> would it be better to chain `Reset`, then `Close` in an helper?

Reset does Close : https://github.com/libp2p/go-yamux/blob/7a85a068c850f401ab7c0177ebafafc959e119bb/stream.go#L282 -> https://github.com/libp2p/go-yamux/blob/7a85a068c850f401ab7c0177ebafafc959e119bb/stream.go#L371
(Close = Close now whatever what and return)
(Reset = Append Close to buffer, wait for ack, then close and then return)

Reply:
Note: `Reset` will _throw away_ any buffered data and signal an error. `Close` will make an effort to deliver the data, but will return before it finishes.

TL;DR: use close when you want to "finish off" a stream. Use `Reset` when you want to abruptly disconnect a stream.

If, instead, you want to _wait_ for any outstanding data to be received, you need to call `CloseWrite()` to flush and send an EOF, then read until you see an EOF from the other side, then finally call `Close()` to actually free resources.